float32 mean_depth
float32 x_min
float32 x_max